sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala

```diff
@@ -107,6 +110,14 @@ class SimpleFunctionRegistry extends FunctionRegistry {
     functionBuilders.remove(name).isDefined
   }
 
+  override def dropMacro(name: String): Boolean = synchronized {
```

Comment:
Hive can drop temporary function using command 'drop Macro'. And it also can  drop temporary macro using command 'drop temporary function'.